torchdata/dataloader2/graph/settings.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import inspect
+
+from typing import Optional
+
+import torch
+
+from torchdata.dataloader2.graph import DataPipe, find_dps, list_dps, traverse_dps
+from torchdata.datapipes.iter import ShardingFilter
+
+
+def _is_random_datapipe(datapipe: DataPipe) -> bool:
+    if hasattr(datapipe, "set_seed") and inspect.ismethod(datapipe.set_seed):
+        return True
+    return False
+
+
+def _generate_random_seed(rng: Optional[torch.Generator] = None, dtype: torch.dtype = torch.int64) -> int:
+    return int(torch.empty((), dtype=dtype).random_(generator=rng).item())
+
+
+def _seed_datapipe(datapipe: DataPipe, seed_generator: torch.Generator) -> DataPipe:
+    seed = _generate_random_seed(seed_generator)
+    datapipe.set_seed(seed)
+    return datapipe
+
+
+def _set_graph_seeds(datapipe: DataPipe, seed_generator: torch.Generator, worker_id: int = 0) -> DataPipe:
```

Comment:
I am using a private name because I don't like the name of this function. LMK if you have any better idea.

Reply:
1. `_set_worker_seed`
2. `_set_worker_seed_for_dp_graph`